internal/core/services/portsprotocol/portsprotocol_fake.go

```diff
@@ -0,0 +1,90 @@
+package portsprotocol
```

Comment:
this is not core. Is simply a transport implementation, what if your app in the future will accept both gRPC and HTTP transport? This should be into package `internal/transport`